tests/pos/i12840/Main.scala

```diff
@@ -0,0 +1,5 @@
+package example
+
+object Main {
+  new NamedImpl() {}
```

Comment:
I suggest making run tests to ensure we're generating valid bytecode and not something that the JVM rejects because we confused an interface for a class